### PR TITLE
ESLintBear: Add filename context

### PR DESCRIPTION
**Pull request by oreqizer**
Adds information about current file location for ESLint.
Aids in relative imports.

Fixes https://github.com/coala-analyzer/coala-bears/issues/741

### CODE REVIEW

**Comment by underyx** [comment]
💥  🎰  💥

File: bears/js/ESLintBear.py [hunk 1]
```diff
@@ -35,7 +35,11 @@ def create_arguments(filename, file, config_file,
         """
         :param eslint_config: The location of the .eslintrc config file.
         """
-        args = '--no-ignore', '--no-color', '-f=json', '--stdin'
+        args = '--no-ignore',\
+               '--no-color',\
+               '-f=json',\
+               '--stdin',\
+               '--stdin-filename=' + filename
```
Comment (underyx) [inline bears/js/ESLintBear.py]: Backslashes 😧 

I wonder why this isn't checked by static analysis checks, using backslashes to escape newlines is considered a pretty bad practice. However!

1. `args = 1, 2, 3` is just syntactic sugar for `args = (1, 2, 3)` (which is creating a `tuple` and is therefore syntactic sugar for `args = tuple(1, 2, 3)`)
2. Newlines don't need to be escaped inside parentheses.

Which means that a multiline
```
args = 1,\
       2,\
       3
```
can also be written as
```
args = (
    1,
    2,
    3,
)
```

**Comment by oreqizer** [comment]
@underyx Fixed. Thanks for the heads up! I'm not really familiar with all of Python syntax and semantics yet.

**Comment by sils1297** [comment]
Hi,

it looks like your PR is out of date and needs a rebase. [This page](http://coala.readthedocs.org/en/latest/Users/Tutorials/Git_Help.html) may help you to get started on this. (*Do not press the update branch button. It's evil.*)

Add the original coala repository as a remote:

```
git remote add upstream https://github.com/coala-analyzer/coala.git
```

Then go to your branch and rebase it onto our master:

```
git checkout <yourbranch>
git fetch upstream
git rebase --interactive upstream/master
```

Now push it up. You will need to do a force push because you change history here (do not do this on master branches :)):

```
git push --force
```

If you're just looking to fix an issue very quickly and not interested in contributing to coala long term, we can fix it up for you - just tell us :).

Ping us on [our gitter channel](https://gitter.im/coala-analyzer/coala) if you need help!

**Comment by sils1297** [comment]
also can we maybe have a test for this? I feel like we defiintely should have one

**Comment by underyx** [comment]
Thanks! That's alright @oreqizer, that's why we're here :D

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/777/commits/613afd9f9c68282f4eda0306c23590351ca51c64

File: bears/js/ESLintBear.py [hunk 2]
```diff
@@ -17,7 +17,9 @@ class ESLintBear:
     """
 
     LANGUAGES = {"JavaScript", "JSX"}
-    REQUIREMENTS = {NpmRequirement('eslint', '2')}
+    REQUIREMENTS = {NpmRequirement('babel-eslint', '6'),
+                    NpmRequirement('eslint', '1'),
+                    NpmRequirement('eslint-plugin-import', '2')}
```
Comment (underyx) [inline bears/js/ESLintBear.py]: How come these versions don't match the `package.json` ones?

Comment (oreqizer) [inline bears/js/ESLintBear.py]: ohh you're right. i'll fix

**Comment by sils1297** [comment]
argh sorry, @AbdealiJK did a lot of bugfixes today and also touched parts of eslintbear which conflicts with your stuff so you'll have to rebase and resolve them. Tell us if you need aything. Sorry :/

**Comment by sils1297** [comment]
I'll pass you a few comments in the meantime

File: bears/js/ESLintBear.py [hunk 3]
```diff
@@ -47,7 +47,12 @@ def create_arguments(filename, file, config_file,
 
     @staticmethod
     def generate_config(filename, file):
-        return '{"extends": "eslint:recommended"}'
+        return """
+        {
+            "extends": "eslint:recommended",
+            "parser": "babel-eslint",
```
Comment (sils1297) [inline bears/js/ESLintBear.py]: I think somebody raised a concern on wether we want this to be default or not but if you think it's safe I'd go with it. Still I'd like you to confirm that once again.

**Comment by sils1297** [comment]
your code looks good if the conflicts wouldn't occur I'd merge it right ahead. Since this is a bugfix I'll wait with the coala brears 0.8.3 release until we merge this :)

**Comment by sils1297** [comment]
marking WIP because of the conflicts though you shouldn't need to change anything

File: bears/js/ESLintBear.py (hunk 3 above)
Comment (oreqizer) [inline bears/js/ESLintBear.py]: It *is* the most common practice, but I am also very hesitant on making it default. I'll revert it during the rebasing just to keep stuff simple.

Comment (sils1297) [inline bears/js/ESLintBear.py]: ok so we probably want to add configurability later and make it a default

**Comment by oreqizer** [comment]
I removed a commit that wasn't a good idea, and squashed the remaining two together, as:

* one was useless without the other
* second one didn't work without the first

so I think they make sense being together.

**Comment by sils1297** [comment]
@oreqizer github still shows conflicts, I think that's why the build didn't start :/

**Comment by oreqizer** [comment]
oh, forgot to fetch upstream.. on it!

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/777/commits/96c86cd75a4b97357cc8f28c2c65796476436e21

**Comment by sils1297** [comment]
@rultor merge

**Pull request merged**